### ISSUE TITLE
FlowQuery FindOperation can't combine `IdentifierFilter` and `AttributeFilters` correctly

### ISSUE THREAD

**Issue by bwaidelich** [neos/neos-development-collection#1923]
### Description

Not a very common usecase, but sometimes one want to filter nodes by their identifier *and*  their type (or other attributes).
According to the documentation this should be possible via a single `FindOperation` like:

```php
$flowQuery->find('#<some-identifier>[instanceof <some-node-type>]')
```
But the `id` filter part is ignored.

### Steps to Reproduce

1. Make sure there are at least two nodes of *SomeType* on one level
2. Take the *identifier* of the *last* node (the one with the highest `sortingIndex`
3. Try to find that node via PHP or Fusion with `find('#<identifier>[instanceof <SomeType>]`).get(0)

#### Expected behavior

The result should point to the *last* node of type *SomeType* with the given Identifier

#### Actual behavior

The result should point to the *first* node of type *SomeType*, ignoring the `#<id>` filter

### Affected Versions

Probably all since `FindOperation` came to life.
Lowest supported branch would be 2.3 as of now

### Background

The source of this issue probably lies in line [152ff of The FindOperation](https://github.com/neos/neos-development-collection/blob/master/Neos.ContentRepository/Classes/Eel/FlowQueryOperations/FindOperation.php#L152) where the attribute filters are just merged with previous id filters

**Comment by kitsunet** [comment]
Yes, technically a bug, but I don't see the point of doing this?

If you know the identifier you can drop all other filters because it's unique. That's like saying I query an entity with this persistence_identifier AND the property value "foo".

**Comment by bwaidelich** [comment]
@kitsunet Here is my usecase: To implement some custom routing I want to find (nested) nodes by their ID but *at the same time* make sure that they are of the suggested type in order to avoid exploits.
Of course I could fetch the nodes first and then check their type, but that's cumbersome and less readable. See related Slack discussion: https://neos-project.slack.com/archives/C050C8FEK/p1519389645000023

**Comment by aertmann** [comment]
Since these filters are fairly tricky to implement, I could see adding support for this not being an easy thing nor pretty code wise. Is it really cumbersome and less readable to add a filter call?

**Comment by bwaidelich** [comment]
Whether it's more or less readable is debatable, but more importantly it is a bug that can sneak in broken behavior quite easily (frankly in a somewhat esotheric scenario).
We even have a [test](https://github.com/neos/neos-development-collection/blob/master/Neos.ContentRepository/Tests/Functional/Eel/FlowQueryOperations/FindOperationTest.php#L41) for that, that doesn't fail because it doesn't test the false positive case.

Personally I'd be OK with just documenting that ID filters are not meant to be combined. I only opened this issue because @hlubek told me to ;)

**Comment by hlubek** [comment]
I see that as a bug and since we do not limit combination of filters (at least from the documented behavior) it's not intuitive that `#identifier[otherfilter]` is not supported.

I had a glance at the code, and it should be doable to support this by checking if we already have a result and then apply subsequent filters by delegating that to `filter()` in `FindOperation`.
